verifiers/plonky2/src/benchmarking.rs

```diff
@@ -0,0 +1,213 @@
+#![cfg(feature = "runtime-benchmarks")]
+
+use crate::Plonky2;
+use frame_benchmarking::v2::*;
+use frame_support::traits::{Consideration, Footprint};
+use frame_system::RawOrigin;
+use hp_verifiers::Verifier;
+use pallet_aggregate::{funded_account, insert_domain};
+use pallet_verifiers::{Tickets, VkEntry, VkOrHash, Vks};
+
+pub struct Pallet<T: Config>(crate::Pallet<T>);
+pub trait Config: crate::Config {}
+impl<T: crate::Config> Config for T {}
+pub type Call<T> = pallet_verifiers::Call<T, Plonky2<T>>;
+
+include!("resources.rs");
+
+fn init<T: pallet_aggregate::Config>() -> (T::AccountId, u32) {
+    let caller: T::AccountId = funded_account::<T>();
+    let domain_id = 1;
+    insert_domain::<T>(domain_id, caller.clone(), Some(1));
+    (caller, domain_id)
+}
+
+#[benchmarks(where T: pallet_verifiers::Config<Plonky2<T>> + pallet_aggregate::Config)]
+mod benchmarks {
```

Comment:
Is there any relevant difference between the Poseidon and Keccak configuration that might be worth to catch in the benchmarks ?

Reply:
If you find it desirable to add, no problem 🙂

Reply:
If you bench locally and see some non-negligible differences, it might be worth to add them, thanks !

Reply:
Also, which proof did you use to run the bench ?
I tried to run benches on our reference machine and it only take 11ms which seems to me a little bit too fast.
Are we able to take a more complex proof from somewhere and use that for benches ?

Reply:
I've using this circuit from `plonky2` repository examples:

https://github.com/NikitaMasych/fibonacci-nostd

Reply:
I'm afraid we should find a more complex real-world example, can you try browsing around a bit to see if you find something ? Thanks !

Reply:
I've searched for some more complex `plonky2` circuit, and here are some considerations:

plonky2 is a type of zkp, which has poly-logarithmic asymptotics on verification, meaning that even for really-really huge circuit, verification speed might increase only negligibly. I've proved it practically, changing my circuit to compute not 100-th element of fibonacci sequence, but rather a 100-million'th - that required me spending appr. 16 minutes on M2 pro with 32 GB ram to prove, but the verification speed changed from 1.1147 ms to 4.7470 ms

Reply:
So, you probably shall not worry that results from small circuit verification times might not be really showcasing plonky2-verifier efficiency

Reply:
I can replace currently used proof in tests/benches in zkVerify-plonky2 to mentioned above with 100-millionth fibonacci though

Reply:
So verifier complexity increase with the degree of the circuit (e.g. num gates)
I was running this bench https://github.com/0xPolygonZero/plonky2/blob/main/plonky2/examples/bench_recursion.rs on my machine and it takes 4-5 ms to verify a proof of degree 2^12 - 2^14 going up to 7-8 ms for a proof of degree 2^19-2^21.
If we want to be over-pedantic, one might want to consider having different benches for different proof degrees (which can be read from `vk.common.degree()`) or just set it to a very big number (like a proof with degree 2^25) and over-estimate the weight (excess fee is reimbursed anyway)

Reply:
I've restricted config to standard one by advice from @lgiussan: https://github.com/zkVerify/zkVerify/pull/194#issuecomment-2654091196.

Hence, we will have a fixed degree.

Reply:
@NikitaMasych actually no, the degree depends on the number of gates in the circuit, regardless of the configuration

Reply:
Ah, my bad, I've confused `FriParams` with `FriConfig`

Reply:
I've set circuit we are using in resources to one with 2^20 degree (got it in [bench_recursion.rs](https://github.com/0xPolygonZero/plonky2/blob/main/plonky2/examples/bench_recursion.rs) - inner one, setting `size` to 20.